core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -32,7 +34,12 @@
  * A dynamic {@link EndpointGroup}. The list of {@link Endpoint}s can be updated dynamically.
  */
 public class DynamicEndpointGroup extends AbstractListenable<List<Endpoint>> implements EndpointGroup {
-    private volatile List<Endpoint> endpoints = ImmutableList.of();
+
+    // An empty list of endpoints we also use as a marker that we have not initialized endpoints yet.
+    private static final List<Endpoint> UNINITIALIZED_ENDPOINTS = Collections.unmodifiableList(
```

Comment:
nit: `ImmutabliList.of()` for consistency?

Reply:
I guess he intentionally used it so that `UNINITIALIZED_ENDPOINTS != ImmutableList.of()` because `ImmutableList.of()` will return a singleton, which means a `DynamicEndpointGroup` will enter the 'unintialized' state again when the endpoint list becomes empty later.

Reply:
Ah, I missed that. Thanks!